src/unstract/sdk/adapters/vectordb/no_op/src/no_op_vectordb.py

```diff
@@ -0,0 +1,80 @@
+import os
+import time
+from typing import Any
+
+from llama_index.core.schema import BaseNode
+from llama_index.core.vector_stores.types import VectorStore
+
+from unstract.sdk.adapters.exceptions import AdapterError
+from unstract.sdk.adapters.vectordb.constants import VectorDbConstants
+from unstract.sdk.adapters.vectordb.helper import VectorDBHelper
+from unstract.sdk.adapters.vectordb.no_op.src.no_op_custom_vectordb import (
+    NoOpCustomVectorDB,
+)
+from unstract.sdk.adapters.vectordb.vectordb_adapter import VectorDBAdapter
+
+
+class NoOpVectorDB(VectorDBAdapter):
+    def __init__(self, settings: dict[str, Any]):
+        self._config = settings
+        self._collection_name: str = VectorDbConstants.DEFAULT_VECTOR_DB_NAME
+        self._vector_db_instance = self._get_vector_db_instance()
+        super().__init__("NoOpVectorDB", self._vector_db_instance)
+
+    @staticmethod
+    def get_id() -> str:
+        return "noOpVectorDb|ca4d6056-4971-4bc8-97e3-9e36290b5bc0"
+
+    @staticmethod
+    def get_name() -> str:
+        return "No Op VectorDB"
+
+    @staticmethod
+    def get_description() -> str:
+        return "No Op VectorDB"
+
+    @staticmethod
+    def get_icon() -> str:
+        return "/icons/adapter-icons/noOpVectorDb.png"
+
+    @staticmethod
+    def get_json_schema() -> str:
+        f = open(f"{os.path.dirname(__file__)}/static/json_schema.json")
+        schema = f.read()
+        f.close()
+        return schema
+
+    def get_vector_db_instance(self) -> VectorStore:
+        return self._vector_db_instance
+
+    def _get_vector_db_instance(self) -> VectorStore:
+        try:
+            dimension = self._config.get(
+                VectorDbConstants.EMBEDDING_DIMENSION,
+                VectorDbConstants.DEFAULT_EMBEDDING_SIZE,
+            )
+            self._collection_name = VectorDBHelper.get_collection_name(
+                self._config.get(VectorDbConstants.VECTOR_DB_NAME),
+                dimension,
+            )
+            vector_db: VectorStore = NoOpCustomVectorDB(
+                dim=dimension, wait_time=self._config.get(VectorDbConstants.WAIT_TIME)
+            )
+            self._client = vector_db.client
+            return vector_db
+        except Exception as e:
+            raise AdapterError(str(e))
+
+    def test_connection(self) -> bool:
```

Comment:
@hari-kuriakose A QQ here - 
Do we need to add a sleep with the configured seconds here just to confirm the sleep happens right in test connection, or should we confine it just to the core methods like completion etc,
![image](https://github.com/user-attachments/assets/a8775967-dbbf-4f55-a10e-2f07154e9c08)
The behavior right now is, returns True immediately, even if the wait is configured. Felt, this might sometimes mislead while testing. But in another way, addition of sleep is not any functionality of this adapter. 
Please advice with your inputs here.

Reply:
@hari-kuriakose Even though it is ok to return immediately in test connection, adding sleep will be better for a consistent UX. Let's do this.